tests/src/cgeo/geocaching/GeocacheTest.java

```diff
@@ -358,4 +359,177 @@ private static Geocache createEventCache(final Calendar calendar) {
+        assertThat(cache.getInventory().get(3).getDistance()).isEqualTo(100F);
+        assertThat(cache.getInventory().get(3).getOwner()).isNull();
+        assertThat(cache.getInventory().get(3).getBrand()).isEqualTo(TrackableBrand.SWAGGIE);
+
+        assertThat(cache.getInventory().get(4)).isEqualTo(trackable7);
+        assertThat(cache.getInventory().get(4).getGeocode()).isEqualTo("GK4321");
+        assertThat(cache.getInventory().get(4).getName()).isEqualTo("GK 4321");
+        assertThat(cache.getInventory().get(4).getDistance()).isEqualTo(300F);
+        assertThat(cache.getInventory().get(4).getOwner()).isNull();
+        assertThat(cache.getInventory().get(4).getBrand()).isEqualTo(TrackableBrand.GEOKRETY);
+    }
+
+    public static void testAddInventoryItem() {
+        final Geocache cache = new Geocache();
+        assertThat(cache.getInventory()).isNull();
+        assertThat(cache.getInventoryItems()).isEqualTo(0);
+
+        // 1st TB
+        final Trackable trackable1 = new Trackable();
+        trackable1.setGeocode("TB1234");
+        trackable1.setName("FOO");
+        trackable1.forceSetBrand(TrackableBrand.TRAVELBUG);
+
+        cache.addInventoryItem(trackable1);
+        assertThat(cache.getInventory()).hasSize(1);
+        assertThat(cache.getInventoryItems()).isEqualTo(1);
+        assertThat(cache.getInventory().get(0).getGeocode()).isEqualTo("TB1234");
+        assertThat(cache.getInventory().get(0).getName()).isEqualTo("FOO");
+        assertThat(cache.getInventory().get(0).getDistance()).isEqualTo(-1F);
+        assertThat(cache.getInventory().get(0).getOwner()).isNull();
+        assertThat(cache.getInventory().get(0).getBrand()).isEqualTo(TrackableBrand.TRAVELBUG);
+
+        // TB to be updated
+        final Trackable trackable2 = new Trackable();
+        trackable2.setGeocode("TB1234");
+        trackable2.setName("BAR");
+        trackable2.setDistance(100);
+        trackable2.forceSetBrand(TrackableBrand.TRAVELBUG);
+        cache.addInventoryItem(trackable2);
+
+        assertThat(cache.getInventory()).hasSize(1);
+        assertThat(cache.getInventoryItems()).isEqualTo(1);
+        assertThat(cache.getInventory().get(0).getGeocode()).isEqualTo("TB1234");
+        assertThat(cache.getInventory().get(0).getName()).isEqualTo("BAR");
+        assertThat(cache.getInventory().get(0).getDistance()).isEqualTo(100F);
+        assertThat(cache.getInventory().get(0).getOwner()).isNull();
+        assertThat(cache.getInventory().get(0).getBrand()).isEqualTo(TrackableBrand.TRAVELBUG);
+
+        // TB to be added
+        final Trackable trackable3 = new Trackable();
+        trackable3.setGeocode("GK6666");
+        trackable3.forceSetBrand(TrackableBrand.GEOKRETY);
+        cache.addInventoryItem(trackable3);
+
+        assertThat(cache.getInventory()).hasSize(2);
+        assertThat(cache.getInventoryItems()).isEqualTo(2);
+        assertThat(cache.getInventory().get(0).getGeocode()).isEqualTo("TB1234");
```

Comment:
I meant the multiple invocations of the same method. You never know whether that called method changes in the future and will take more runtime. Also you never know if the method will return the same value on second, third,... invocation. And therefore you should _never_ call the same method twice in the same scope, but instead assign to a local variable. That's the most easy rule for defensive and speedy programming.

Will merge nevertheless.